app/models/comment.php

```diff
@@ -42,6 +63,72 @@ public function write()
         );
 
         $db = DB::conn();
-        $db->insert('comment',$params);
+        $db->insert(self::TABLE,$params);
+    }
+
+    public static function getByThreadId($thread_id)
+    {
+        $db = DB::conn();
+        return $db->row(
+            sprintf("SELECT * FROM comment WHERE thread_id = ? LIMIT %d", self::FIRST_COMMENT),
+            array($thread_id)
+        );
+    }
+
+    public static function getIdByThreadId($thread_id)
+    {
+        $db = DB::conn();
+        return $db->value(
+            sprintf("SELECT id FROM comment WHERE thread_id = ? LIMIT %d", self::FIRST_COMMENT),
+            $thread_id
+        );
+    }
+
+    public function edit()
+    {
+        if(!$this->validate()) {
+            throw new ValidationException;
+        }
+
+        $db = DB::conn();
+        $db->query(
+            "UPDATE comment SET body=?, last_modified=NOW() WHERE id=? AND user_id = ?",
+            array($this->body, $this->id, $_SESSION['userid'])
+        );
+    }
+
+    public function isAuthor()
+    {
+        $db = DB::conn();
+        $params = array(
+            $this->id,
+            $_SESSION['userid']
```

Comment:
Sessions should be in the controller. 
Refer to [this:]( https://github.com/maragatchalian/FoodFacts/commit/9ac61e6db6d7e57f5ab8791330210737c5c13687#diff-543f1721d9208970049520efce19dd5aR236)


Apply to all parts of your code :smile: